torch/testing/_internal/common_device_type.py

```diff
@@ -163,7 +165,10 @@
 # List of device type test bases that can be used to instantiate tests.
 # See below for how this list is populated. If you're adding a device type
 # you should check if it's available and (if it is) add it to this list.
-device_type_test_bases = []
+
+#TDeviceTypeTestBase = TypeVar('TDeviceTypeTestBase', bound=DeviceTypeTestBase)
+# todo: need to fix
+device_type_test_bases : typing.List[typing.Union[CPUTestBase, CUDATestBase]] = []
```

Comment:
can you try just use the base class: `TDeviceTypeTestBase` instead of the union?